CONTRIBUTING.rst

```diff
@@ -7,373 +7,372 @@ Contributing
-          entry that explains the changes that will be included in the new version.
-          Normally this is just a list of the Pull Requests that have been merged to master
-          since the last release.
+1. ``generate`` takes in the number of rows to generate, and outputs the expected number
+   of data rows.
+2. ``get_performance_thresholds`` returns the time and memory threshold for each of the required
+   transformer methods. These thresolds are per row.
 
-Once this is done, run of the following commands:
+You should make a generator for every type of column that you believe would be useful to test
+against. For some examples, you can look in this
+folder: https://github.com/sdv-dev/RDT/tree/master/tests/datasets
 
-1. If you are releasing a patch version::
+The generators each have a ``DATA_TYPE`` class variable. This should match the data type that your
+``transformer`` accepts as input.
 
-    make release
+Common Performance Pitfalls
+"""""""""""""""""""""""""""
+
+It is important to keep the performance of these transformers as efficient as possible.
+Below are some tips and common pitfalls to avoid when developing your transformer, so as to
+optimize performance.
+
+1. Avoid duplicate operations. If you need to do some change to an array/series, try to only
+   do it once and reuse that variable later.
+2. Try to use vectorized operations when possible.
+3. When working with Pandas Series, a lot of the operations are able to handle nulls. If you
+   need to round, get the max or get the min of a series, there is no need to filter out nulls
+   before doing that calculation.
+4. ``pd.to_numeric`` is preferred over ``as_type``.
+5. ``pd.to_numeric`` also replaces all None values with NaNs that can be operated on since
+   ``np.nan`` is a float type.
+6. If you are working with a series that has booleans and null values, there is a
+   `nullable boolean type`_ that can be leveraged to avoid having to filter out null values.
 
-2. If you are releasing a minor version::
+Validating Performance
+""""""""""""""""""""""
 
-    make release-minor
+Validate the performance of your transformer using the
+``validate_transformer_performance`` function::
 
-3. If you are releasing a major version::
+    from tests.contributing import validate_transformer_performance
+    validate_transformer_performance('rdt.transformers.<YourTransformer>')
 
-    make release-major
+Fix any performance issues that are reported. If there are no errors but performance
+can be improved, this function should be used for reference.
 
-Release Candidates
-~~~~~~~~~~~~~~~~~~
+Transformer Quality
+~~~~~~~~~~~~~~~~~~~
 
-Sometimes it is necessary or convenient to upload a release candidate to PyPi as a pre-release,
-in order to make some of the new features available for testing on other projects before they
-are included in an actual full-blown release.
+To assess the quality of a transformer, we run quality tests that apply the Transformer
```

Comment:
Do we need to tell them to skip this step if it's not applicable (e.g. if the transformer isn't expected to preserve relationships).

I remember there was some way to denote that a transformer should be skipped for quality testing.